Improve the performance of the project

zgx949 opened the pull request:
Added GPU and other device inference
Improve the shooting hit detection algorithm

zgx949 committed:
perf: added `get_device` for training speed up

zgx949 committed:
perf: added a larger dataset url

zgx949 committed:
fix: `ball_pos[i+1]` index out of range

zgx949 committed:
perf: Hit detection & accelerated inference

zgx949 committed:
docx: added Chinese version

zgx949 committed:
added: pre-trained model & test video

zgx949 committed:
added: Shooting evaluation indicators

avishah3 reviewed (commented):
Hi, thank you for making all these changes. Before I merge them, could you please explain the purpose of the simulate rebound path functionality? Also, could you let me know how accurate this part is during your testing?

Thanks.

zgx949 commented:
@avishah3 

Sorry, the current rebound detection effect is not very good. The initial idea was to determine 'Good' when the basketball hits the basket and re enters the basket, and 'Perfect' when it enters the hollow basket. How about I temporarily delete the code for rebound detection?

zgx949 commented:
Other changes mainly include the following:
1. I switched to a larger dataset because the dataset you originally provided was relatively small, and the model I trained did not perform particularly well, especially when the images were relatively small. I have already trained a model using a big dataset for you to test.
2. Optimized the algorithm of the 'score' function, as there may be array index overflow in the original 'ball_pos [i+1]'.
3. Added automatic recognition GPU devices to facilitate accelerated training and inference.
4. Added Chinese documents
5. Added trained model files and testing video files (convenient for those who want to test project results and those who don't have time to train)
